plugin/plugin.go

```diff
@@ -154,8 +154,22 @@ func (p *plugin) generateRegexVars(file *generator.FileDescriptor, message *gene
 	ccTypeName := generator.CamelCaseSlice(message.TypeName())
 	for _, field := range message.Field {
 		validator := getFieldValidatorIfAny(field)
-		if validator != nil && validator.Regex != nil {
+		if validator != nil && (validator.Regex != nil || validator.UuidVer != nil) {
 			fieldName := p.GetOneOfFieldName(message, field)
+			if validator.UuidVer != nil {
+				if uuid, err := getUUIDRegex(int(validator.GetUuidVer())); err != nil {
```

Comment:
In order to reduce the high nesting level of this piece of code I would suggest making the new `getUUIDRegex` directly take the `validator.UuidVer` pointer and return a pointer to a string (`nil` if the input pointer is `nil`). That way this could roughly be rewritten as:

```go
if validator != nil && (validator.Regex != nil || validator.UuidVer != nil) {
	if validator.Regex != nil && validator.UuidVer != nil {
		// print warning
	}

	fieldName := p.GetOneOfFieldName(message, field)
	if uuidRegex, err := getUUIDRegex(validator.UuidVer); err != nil {
		// print warning
	} else if uuidRegex != nil {
		validator.Regex = uuidRegex
	}
	// Generated code printing...
}
```

Reply:
I restructured the loop, to reduce the nesting.
Please have a look again.